### PR TITLE
feat(table): add options to allow customization of cell renderers

### PR DESCRIPTION
**Pull request by adisreyaj**
## Description

- Add extra `options` prop to the `TableColumnConfig`
- Use it to pass extra data to the renderers which can then be used for customizing the renderers.

### CODE REVIEW

File: projects/components/src/table/table-api.ts [hunk 1]
```diff
@@ -4,20 +4,35 @@ import { FieldFilter, FilterValue } from '../filtering/filter/filter';
 import { FilterOperator } from '../filtering/filter/filter-operators';
 import { TableCellAlignmentType } from './cells/types/table-cell-alignment-type';
 
-export interface TableColumnConfig {
-  id: string; // This is the unique ID for the column (often same as 'name' except for composite fields)
-  name?: string; // Attribute name (for composite columns use the attribute that should be filtered/sorted)
+export interface TableColumnConfig<TableColumnOptions = unknown> {
```
Comment (arjunlalb) [inline projects/components/src/table/table-api.ts:7]: I think its safe to assume `Dictionary<unknown>` as type.

**Review by arjunlalb — DISMISSED** [review]

File: projects/components/src/table/cells/table-cell-injection.ts [hunk 2]
```diff
@@ -36,6 +37,10 @@ export const createTableCellInjector = (
         provide: TABLE_CELL_DATA,
         useValue: cellData
       },
+      {
+        provide: TABLE_CELL_OPTIONS,
```
Comment (anandtiwary) [inline projects/components/src/table/cells/table-cell-injection.ts:41]: Why do we need to inject it? Column config is already injected, can we pull from there?

Comment (adisreyaj) [reply projects/components/src/table/cells/table-cell-injection.ts:41]: Yeah. I think will make the change then.

File: projects/components/src/table/table-api.ts (hunk 1 above)
Comment (adisreyaj) [reply projects/components/src/table/table-api.ts:7]: Started off with that, but had type conflicts downstream.

Comment (anandtiwary) [reply projects/components/src/table/table-api.ts:7]: Could we do `TableColumnOptions = Dictionary<boolean>` so that we only allow a bunch of booleans?

Comment (adisreyaj) [reply projects/components/src/table/table-api.ts:7]: There can be some other cases where passing enum would also make sense for options.

Comment (arjunlalb) [reply projects/components/src/table/table-api.ts:7]: agree with possible need for enums. but why wouldn't `TableColumnOptions extends Dictionary<unknown>` type work?

**Review by arjunlalb — APPROVED** [review]

File: projects/components/src/table/table-api.ts (hunk 1 above)
Comment (adisreyaj) [reply projects/components/src/table/table-api.ts:7]: I can come around to it in some time.

**Pull request merged**